### ISSUE TITLE
stop using `terraform apply -target`

### ISSUE THREAD

**Issue by arcnmx** [arcnmx/tf-nix#28]
Partial applies have been deprecated by terraform for a while now, and may be removed in the future. The obvious solution is to:

- invert our polarity: instead of having a nix derivation execute terraform, generate HCL for a resource that evaluates nix
- make all inputs that go from HCL into the nix evaluation explicit
  - ... so that all that the `deps` module needs to do is auto-fill this list of inputs
- allow nix evals to have outputs that are later used in HCL and/or future resource evaluations

basically, everything that is currently a recursive execution now just becomes one or more nix eval resources in the output HCL - and it may be possible to get it down to one in most/all cases?
